src/client/scripts/game/chess/variant.js

```diff
@@ -55,10 +55,16 @@ const variant = (function() {
      */
     function initExistingTypes(gamefile) {
         const teamtypes = new Set(Object.values(gamefile.startSnapshot.position)); // Make a set of all pieces in game
```

Comment:
Can this make use of gamefile.startSnapshot.existingTypes? Or is that not defined yet by this point?